### PR TITLE
add vpa CRs

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
Towards https://github.com/giantswarm/giantswarm/issues/28808

Add vpa CRs to the templates.
Currently under testing

### CODE REVIEW

File: helm/keda/README.md [hunk 1]
```diff
@@ -117,8 +117,7 @@ Kubernetes: `>=v1.24.0-0`
 | prometheus.metricServer.serviceMonitor.jobLabel | string | `nil` |  |
 | prometheus.metricServer.serviceMonitor.podTargetLabels | list | `[]` |  |
 | prometheus.metricServer.serviceMonitor.port | string | `"metrics"` |  |
-| prometheus.metricServer.serviceMonitor.relabellings[0].replacement | string | `"keda"` |  |
```
Comment (QuentinBisson) [inline helm/keda/README.md]: Something looks wrong here

File: helm/keda/README.md [hunk 2]
```diff
@@ -157,8 +155,7 @@ Kubernetes: `>=v1.24.0-0`
 | prometheus.webhooks.serviceMonitor.jobLabel | string | `nil` |  |
 | prometheus.webhooks.serviceMonitor.podTargetLabels | list | `[]` |  |
 | prometheus.webhooks.serviceMonitor.port | string | `"metrics"` |  |
-| prometheus.webhooks.serviceMonitor.relabellings[0].replacement | string | `"keda"` |  |
-| prometheus.webhooks.serviceMonitor.relabellings[0].targetLabel | string | `"app"` |  |
+| prometheus.webhooks.serviceMonitor.relabellings | list | `[]` |  |
```
Comment (QuentinBisson) [inline helm/keda/README.md]: Also we should have the verticalPodAutoscaler section

Comment (QuentinBisson) [reply helm/keda/README.md]: Did you run 'helm-docs helm/keda' and schema-gen ?

Comment (QuantumEnigmaa) [reply helm/keda/README.md]: It's strange because I din't change this manually. It was done by vendir

Comment (QuentinBisson) [reply helm/keda/README.md]: That's probably because I did not run helm-doc in keda-upstream

Comment (QuantumEnigmaa) [reply helm/keda/README.md]: > Did you run 'helm-docs helm/keda' and schema-gen ?

I did run `helm-docs` but forgot `schema-gen`

Comment (QuentinBisson) [reply helm/keda/README.md]: Can you run it again? Because I get different output

Comment (QuantumEnigmaa) [reply helm/keda/README.md]: Seems better I think

**Comment by QuantumEnigmaa** [comment]
@QuentinBisson can you update the schema ? the following command : `helm schema-gen helm/keda/values.yaml > helm/keda/values.schema.json` gives an empty file and the output is too large to be copied manually from the terminal :/

**Comment by QuentinBisson** [comment]
Done :)

**Review by QuentinBisson — APPROVED** [review]
LGTM if tested

File: helm/keda/values.yaml [hunk 3]
```diff
@@ -35,6 +35,24 @@ image:
 # Kubernetes cluster domain
 clusterDomain: cluster.local
 
+verticalPodAutoscaler:
+  keda:
+    enabled: false
```
Comment (nprokopic) [inline helm/keda/values.yaml]: Would we ever have VPA disabled? Why not just render VPA unconditionally?

Comment (QuentinBisson) [reply helm/keda/values.yaml]: Oh in our values that make sense yes

Comment (QuantumEnigmaa) [reply helm/keda/values.yaml]: I updated the values. Now vpa is enabled by default

Comment (QuentinBisson) [reply helm/keda/values.yaml]: Did you update that in keda-upstream branches first then apply vendir?

Comment (QuantumEnigmaa) [reply helm/keda/values.yaml]: Oh no my bad. Let me fix it

Comment (QuantumEnigmaa) [reply helm/keda/values.yaml]: Done

**Comment by QuantumEnigmaa** [comment]
Tested on `gaia` and `gauss` and seems like it's working : 
```
zirco@Zirko-laptop:~/keda/keda-app$ k get pods -A | grep
keda              keda-admission-webhooks-7885988d78-fbtfn                              1/1     Running     0             17m
keda              keda-operator-5f8ff4cd9d-9wpqm                                        1/1     Running     0             17m
keda              keda-operator-metrics-apiserver-77755b4699-phd29                      1/1     Running     0             17m
zirco@Zirko-laptop:~/keda/keda-app$ k get vpa -n keda
NAME                                  MODE   CPU   MEM       PROVIDED   AGE
keda-operator-metrics-apiserver-vpa   Auto   50m   262144k   True       17m
keda-operator-vpa                     Auto   50m   262144k   True       17m
```

**Comment by QuentinBisson** [comment]
Awesome :)

**Comment by QuentinBisson** [comment]
I think we're ready for a release then :) and a new pr upstream :)

**Comment by QuantumEnigmaa** [comment]
yeah !

**Pull request merged**